x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicators_field_selector/indicators_field_selector.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useState } from 'react';
+import { EuiSelect, EuiSelectOption } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DataView, DataViewField } from '@kbn/data-views-plugin/common';
+import { RawIndicatorFieldId } from '../../../../../common/types/indicator';
+
+export const DROPDOWN_TEST_ID = 'tiIndicatorFieldSelectorDropdown';
+
+export interface IndicatorsFieldSelectorProps {
+  indexPatterns: DataView[];
+  valueChange: (value: string) => void;
+  defaultStackByValue?: RawIndicatorFieldId;
+}
+
+const DEFAULT_STACK_BY_VALUE = RawIndicatorFieldId.Feed;
+
+export const IndicatorsFieldSelector = memo<IndicatorsFieldSelectorProps>(
+  ({ indexPatterns, valueChange, defaultStackByValue = DEFAULT_STACK_BY_VALUE }) => {
+    const [selectedField, setSelectedField] = useState<string>(defaultStackByValue);
+
+    if (indexPatterns && indexPatterns.length === 0) return <></>;
+    const fields: EuiSelectOption[] = indexPatterns[0].fields.map((f: DataViewField) => ({
+      text: f.name,
+      value: f.name,
+    }));
+
+    return (
+      <EuiSelect
+        data-test-subj={DROPDOWN_TEST_ID}
+        onChange={(event) => {
+          valueChange(event.target.value);
+          setSelectedField(event.target.value);
+        }}
```

Comment:
`useCallback` and `memo`izing the non-primitive props will prevent `<EuiSelect />` from re-rendering it's internals every time IndicatorFieldsSelector is updated

Reply:
done!